src/Common/src/CoreLib/System/IO/Stream.cs

```diff
@@ -758,17 +758,13 @@ public virtual int Read(Span<byte> buffer)
 
         // Reads one byte from the stream by calling Read(byte[], int, int). 
         // Will return an unsigned byte cast to an int or -1 on end of stream.
-        // This implementation does not perform well because it allocates a new
-        // byte[] each time you call it, and should be overridden by any 
-        // subclass that maintains an internal buffer.  Then, it can help perf
-        // significantly for people who are reading one byte at a time.
         public virtual int ReadByte()
         {
-            byte[] oneByteArray = new byte[1];
-            int r = Read(oneByteArray, 0, 1);
+            Span<byte> oneByteSpan = stackalloc byte[1];
+            int r = Read(oneByteSpan);
```

Comment:
I'm not convinced this is an improvement (which is why I've avoided making the same change). If someone has their own stream and has overridden the new span-based overloads, seems likely they've also overridden ReadByte, in which case this won't matter. And if they haven't overridden the span-based overload, this implementation is likely to be worse-performing than the previous one.

Reply:
Did you run any benchmarks when you made the decision? I could throw some together and see what happens speed/allocations wise.

Reply:
Reverted in https://github.com/dotnet/corefx/pull/36838/commits/1daf838d3f5a474856e9c3503604ba37758ad8fc